kernel_tuner/convert/convert.py

```diff
@@ -0,0 +1,102 @@
+import json
+from pathlib import Path
+
+# The path construction here probably needs to be changed when this file becomes part of the library
+PROJECT_DIR = Path(__file__).parents[0]
+
+SCHEMA_VERSIONS_PATH = PROJECT_DIR / "../schema/cache/convert_temp"
+
+DEFAULT_VALUES = {
+    "object":   dict(),
+    "array":    list(),
+    "string":   "default string",
+    "integer":  0,
+    "number":   0,
+    "true":     True,
+    "false" :   False,
+    "null":     None
+}
+
+
+def convert_cache_file(filestr):
+    # Get the version of the cachefile
+    with open(filestr, 'r') as cachefile:
+        cache = json.load(cachefile)
+        version = cache["schema_version"]
+
+    orig_version = version    
+
+    # Obtain a sorted list of all versions
+    versions = list(map (lambda v:v.name, sorted(SCHEMA_VERSIONS_PATH.iterdir())))
+
+    func_list = globals()
+
+    for v in range(len(versions)-1):
+        oldver = versions[v]
+        newver = versions[v+1]
+
+        if oldver == version:
+            # Convert cachefile from oldver to newver
+
+            with open(filestr, 'r') as cachefile:
+                cache = json.load(cachefile)
+
+            # Look first for an implemented conversion function
+            # If no such function exists, attempt default conversion
+            func = ("c" + oldver + "_to_" + newver).replace(".", "_")
+
+            if func in func_list:
+                cache = func_list[func](cache)
+            else:
+                cache = default_convert(cache, oldver, newver)
+
+            version = newver
+            cache["schema_version"] = version
+
+            with open(filestr, 'w') as cachefile: 
+                cachefile.write(json.dumps(cache, indent=4))
+
+
+    print("Cache file '{}' converted from version {} to version {}".format(filestr, orig_version, version))
```

Comment:
Als deze functie vanuit python wordt aangeroepen, dan wordt ook geprint. Ik kan het mij voorstellen dat wanneer je de converter van de command line aanroept zo'n print statement handig is. Is het ook de bedoeling dat altijd dit wordt geprint?